dipy/reconst/csdeconv.py

```diff
@@ -118,18 +120,20 @@ def __init__(self, gtab, response, reg_sphere=None, sh_order=8, lambda_=1,
         # scale lambda_ to account for differences in the number of
         # SH coefficients and number of mapped directions
         # This is exactly what is done in [4]_
-        self.lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
-                        (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.))
-                       )
+
+        lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
+                   (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.)))
```

Comment:
On Tue, Jan 27, 2015 at 2:09 PM, Samuel St-Jean <notifications@github.com>
wrote:

> In dipy/reconst/csdeconv.py
> <https://github.com/nipy/dipy/pull/553#discussion_r23647669>:
>
> > @@ -118,18 +120,20 @@ def __init__(self, gtab, response, reg_sphere=None, sh_order=8, lambda_=1,
> >          # scale lambda_ to account for differences in the number of
> >          # SH coefficients and number of mapped directions
> >          # This is exactly what is done in [4]_
> > -        self.lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
> > -                        (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.))
> > -                       )
> > +
> > +        lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
> > +                   (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.)))
>
> Actually sqrt(len(self.sphere.vertices)) (or something like that), so the
> default code would still stay at sqrt(362), and other sphere would scale
> the value accordingly.
>
It's in the code,  `self.B_reg.shape[0] == len(self.sphere.vertices)`.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/553/files#r23647669>.
>